File: frontend/src/types/news.ts
/** News and anomaly types for v2.0. */

export interface StockNewsItem {
  title: string
  content: string
  datetime: string
  source: string
  url: string
  sentiment?: "positive" | "negative" | "neutral"
  impact_level?: "high" | "medium" | "low"
}

export interface AnomalyItem {
  datetime: string
  symbol: string
  name: string
  change_type: string
  description: string
  sector?: string
}

export interface SentimentSummary {
  symbol: string
  overall: "positive" | "negative" | "neutral"
  positive_count: number
  negative_count: number
  neutral_count: number
  total_count: number
  score: number
  summary: string | null
}

export interface HotRankItem {
  rank: number
  symbol: string
  name: string
  price: number | null
  pct_change: number | null
}

File: src/web/schemas/news.py
"""News and sentiment Pydantic models."""

from __future__ import annotations

from pydantic import BaseModel, Field


class StockNewsItem(BaseModel):
    """Stock news article."""

    title: str
    content: str = ""
    datetime: str = ""
    source: str = ""
    url: str = ""
    sentiment: str | None = None
    impact_level: str | None = None


class AnomalyItem(BaseModel):
    """Stock anomaly/unusual activity record."""

    datetime: str = ""
    symbol: str = ""
    name: str = ""
    change_type: str = ""
    description: str = ""
    sector: str | None = None


class SentimentSummary(BaseModel):
    """Aggregated news sentiment summary."""

    symbol: str
    overall: str = "neutral"
    positive_count: int = 0
    negative_count: int = 0
    neutral_count: int = 0
    total_count: int = 0
    score: float = 0.0
    summary: str | None = None


class ResearchResult(BaseModel):
    """Aggregated research data for a stock."""

    symbol: str = ""
    news: list[StockNewsItem] = Field(default_factory=list)
    sentiment: SentimentSummary | None = None
    fund_holdings: list[dict] = Field(default_factory=list)
    analyst_ratings: list[dict] = Field(default_factory=list)
